File: test_ollama_api.py
#!/usr/bin/env python3
import requests
import json

def test_ollama_direct():
    """測試直接調用 Ollama"""
    print("🔧 測試直接調用 Ollama...")
    
    url = "http://localhost:11434/api/generate"
    payload = {
        "model": "qwen2.5:7b",
        "prompt": "你好，請分析當前交通狀況並提供建議。",
        "stream": False,
        "options": {
            "temperature": 0.7,
            "num_predict": 200
        }
    }
    
    try:
        response = requests.post(url, json=payload, timeout=30)
        if response.ok:
            result = response.json()
            print("✅ Ollama 直接調用成功!")
            print(f"回應: {result.get('response', '無回應')[:100]}...")
            return True
        else:
            print(f"❌ Ollama 調用失敗: {response.status_code}")
            return False
    except Exception as e:
        print(f"❌ Ollama 連接錯誤: {e}")
        return False

def test_traffic_api():
    """測試交通數據 API"""
    print("\n🚗 測試交通數據 API...")
    
    try:
        response = requests.get("http://localhost:8000/api/traffic/current", timeout=10)
        if response.ok:
            data = response.json()
            print(f"✅ 交通數據獲取成功! 站點數: {len(data.get('stations', []))}")
            return data
        else:
            print(f"❌ 交通數據獲取失敗: {response.status_code}")
            return None
    except Exception as e:
        print(f"❌ 交通數據 API 錯誤: {e}")
        return None

def test_integrated_chat():
    """測試整合的聊天功能"""
    print("\n🤖 測試整合聊天功能...")
    
    # 獲取交通數據
    traffic_data = test_traffic_api()
    
    # 模擬交通數據分析請求
    url = "http://localhost:11434/api/generate"
    
    # 構建包含交通數據的提示
    if traffic_data:
        stations = traffic_data.get('stations', [])
        total_stations = len(stations)
        avg_speed = sum(s.get('speed', 0) for s in stations) / total_stations if total_stations > 0 else 0
        
        traffic_summary = f"""
當前交通狀況：
- 監測站點: {total_stations} 個
- 平均車速: {avg_speed:.1f} km/h
- 更新時間: {traffic_data.get('last_updated', '未知')}
"""
        
        prompt = f"""你是一個專業的交通分析助手，請根據以下即時交通數據提供駕駛建議：

{traffic_summary}

用戶問題：目前路況如何？有什麼駕駛建議？

請用繁體中文回答，提供具體且實用的建議。"""
    else:
        prompt = "你是交通助手，目前沒有即時數據，請提供一般性的駕駛安全建議。"
    
    payload = {
        "model": "qwen2.5:7b",
        "prompt": prompt,
        "stream": False,
        "options": {
            "temperature": 0.7,
            "num_predict": 300
        }
    }
    
    try:
        response = requests.post(url, json=payload, timeout=30)
        if response.ok:
            result = response.json()
            print("✅ 整合聊天測試成功!")
            print("🤖 AI 回應:")
            print(result.get('response', '無回應'))
            return True
        else:
            print(f"❌ 整合聊天失敗: {response.status_code}")
            return False
    except Exception as e:
        print(f"❌ 整合聊天錯誤: {e}")
        return False

if __name__ == "__main__":
    print("🚀 開始測試 Ollama 整合功能...\n")
    
    # 測試各個組件
    ollama_ok = test_ollama_direct()
    traffic_ok = test_traffic_api() is not None
    chat_ok = test_integrated_chat()
    
    print(f"\n📊 測試結果總結:")
    print(f"- Ollama 服務: {'✅' if ollama_ok else '❌'}")
    print(f"- 交通數據 API: {'✅' if traffic_ok else '❌'}")
    print(f"- 整合聊天功能: {'✅' if chat_ok else '❌'}")
    
    if ollama_ok and traffic_ok and chat_ok:
        print("\n🎉 所有測試通過！可以開始整合前端。")
    else:
        print("\n⚠️ 部分測試失敗，需要修復後再繼續。")

File: test-ollama-api.js
#!/usr/bin/env node

/**
 * Ollama API 測試腳本
 * 用於診斷 Next.js API Route 與 Ollama 的連接問題
 */

const fetch = require('node-fetch');

const OLLAMA_URL = 'http://localhost:11434';
const NEXT_API_URL = 'http://localhost:3000';

async function testOllamaDirectly() {
  console.log('\n🔵 測試 1: 直接連接 Ollama 服務');
  console.log('URL:', OLLAMA_URL);

  try {
    // 測試 /api/tags
    const tagsResponse = await fetch(`${OLLAMA_URL}/api/tags`);
    if (tagsResponse.ok) {
      const data = await tagsResponse.json();
      console.log('✅ Ollama 服務正常');
      console.log('可用模型:', data.models.map(m => m.name).join(', '));

      // 測試 /api/generate
      console.log('\n測試生成回應...');
      const generateResponse = await fetch(`${OLLAMA_URL}/api/generate`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          model: 'qwen2.5:7b',
          prompt: '你好，請簡單回答',
          stream: false
        })
      });

      if (generateResponse.ok) {
        const result = await generateResponse.json();
        console.log('✅ 生成測試成功');
        console.log('回應:', result.response.substring(0, 100) + '...');
      } else {
        console.error('❌ 生成測試失敗:', generateResponse.status);
      }
    } else {
      console.error('❌ Ollama 服務無回應:', tagsResponse.status);
    }
  } catch (error) {
    console.error('❌ 連接 Ollama 失敗:', error.message);
  }
}

async function testNextAPI() {
  console.log('\n🔵 測試 2: Next.js API Route');
  console.log('URL:', `${NEXT_API_URL}/api/rag/chat`);

  try {
    // 測試 GET (健康檢查)
    const healthResponse = await fetch(`${NEXT_API_URL}/api/rag/chat`);
    if (healthResponse.ok) {
      const data = await healthResponse.json();
      console.log('✅ API 健康檢查成功');
      console.log('狀態:', data.status);
    } else {
      console.error('❌ API 健康檢查失敗:', healthResponse.status);
      const errorText = await healthResponse.text();
      console.error('錯誤:', errorText);
    }

    // 測試 POST (對話)
    console.log('\n測試 POST 請求...');
    const chatResponse = await fetch(`${NEXT_API_URL}/api/rag/chat`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        message: '你好，請簡單回答',
        user_location: { lat: 25.0330, lng: 121.5654 }
      })
    });

    if (chatResponse.ok) {
      const data = await chatResponse.json();
      console.log('✅ API 對話測試成功');
      console.log('回應:', data.response.substring(0, 100) + '...');
    } else {
      console.error('❌ API 對話測試失敗:', chatResponse.status);
      const errorData = await chatResponse.json();
      console.error('錯誤:', JSON.stringify(errorData, null, 2));
    }
  } catch (error) {
    console.error('❌ 連接 Next.js API 失敗:', error.message);
    console.error('請確認開發伺服器是否在運行 (npm run dev)');
  }
}

async function main() {
  console.log('═══════════════════════════════════════');
  console.log('  Ollama API 診斷工具');
  console.log('═══════════════════════════════════════');

  await testOllamaDirectly();
  await testNextAPI();

  console.log('\n═══════════════════════════════════════');
  console.log('  診斷完成');
  console.log('═══════════════════════════════════════\n');
}

main().catch(console.error);
